ideascube/serveradmin/models.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+
+class Setting(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def _get_typed(cls, namespace, key, type, default=None):
+        try:
+            setting = Setting.objects.get(namespace=namespace, key=key)
+
+        except Setting.DoesNotExist:
+            if default is not None:
+                return default
+
+            raise
+
+        else:
+            if isinstance(setting.value, type):
+                return setting.value
+
+            if default is not None:
+                return default
+
+            raise TypeError(
+                '%s is not a %s setting' % (setting, type.__name__))
+
+    @classmethod
+    def get_string(cls, namespace, key, default=None):
+        return cls._get_typed(namespace, key, str, default=default)
+
+    @classmethod
+    def set(cls, namespace, key, value, actor):
+        try:
```

Comment:
Why not using https://docs.djangoproject.com/en/1.9/ref/models/querysets/#update-or-create ?